runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -119,16 +118,11 @@ protected boolean discoverAddressSpace(final UUID streamId,
                 log.trace("followBackpointers: address[{}] contains streamId[{}], apply filter", currentAddress,
                         streamId);
                 // Check whether we should include the address
-                BackpointerOp op = filter.apply(d);
-                if (op == BackpointerOp.INCLUDE
-                        || op == BackpointerOp.INCLUDE_STOP) {
+                filter.apply(d);
+
+                if (!checkpoint) {
                     log.trace("followBackpointers: Adding backpointer to address[{}] to queue", currentAddress);
                     queue.add(currentAddress);
-                    entryAdded = true;
-                    // Check if we need to stop
-                    if (op == BackpointerOp.INCLUDE_STOP) {
-                        return entryAdded;
```

Comment:
You can remove this variable now, it always false.

Reply:
Done. Actually returning the state of a queue cause entries could have been added before hitting a trimmed exception in back pointer implementation.